boxes/generators/dicebox.py

```diff
@@ -0,0 +1,112 @@
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+
+
+class DiceBox(Boxes):
+    """Box with lid and integraded hinge for storing dice"""
+
+    ui_group = "Misc"
+
+    def __init__(self):
+        Boxes.__init__(self)
+        self.addSettingsArgs(
+            edges.FingerJointSettings,
+            surroundingspaces=2.0)
+        self.addSettingsArgs(
+            edges.ChestHingeSettings,
+            finger_joints_on_box=True,
+            finger_joints_on_lid=True)
+        self.buildArgParser(
+            x=100,
+            y=100,
+            h=18,
+            outside=True)
+        self.argparser.add_argument(
+            "--lidheight",  action="store", type=float, default=18,
+            help="height of lid in mm")
+        self.argparser.add_argument(
+            "--hex_hole_corner_radius",  action="store", type=float, default=5,
+            help="The corner radius of the hexagonal dice holes, in mm")
+        self.argparser.add_argument(
+            "--magnet_diameter",  action="store", type=float, default=6,
+            help="The diameter of magnets for holding the box closed, in mm")
+
+    def diceCB(self):
+        t = self.thickness
+        xc = self.y / 2 - t
+        yc = self.x / 2 - t
+        cr = self.hex_hole_corner_radius
+
+        r = (min(self.x, self.y) - 2 * t) / 2 / math.pi
+        apothem = r * math.sqrt(3) / 2
+
+        # dice
+        self.regularPolygonHole(xc, yc, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc, yc+2*apothem+t, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc+2*apothem, yc+apothem+t/2, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc-2*apothem, yc+apothem+t/2, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc, yc-2*apothem-t, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc+2*apothem, yc-apothem-t/2, n=6, r=r, corner_radius=cr)
+        self.regularPolygonHole(xc-2*apothem, yc-apothem-t/2, n=6, r=r, corner_radius=cr)
+
+        # magnets
+        d = self.magnet_diameter
+        self.hole(t+d/2, t+d/2, d=d)
+        self.hole(t+d/2, self.x-3*t-d/2, d=d)
+
+    def render(self):
+        x, y, h, hl = self.x, self.y, self.h, self.lidheight
+
+        if self.outside:
+            x = self.adjustSize(x)
+            y = self.adjustSize(y)
+            h = self.adjustSize(h)
+            hl = self.adjustSize(hl)
+
+        t = self.thickness
+
+        hy = self.edges["O"].startwidth()
+        hy2 = self.edges["P"].startwidth()
+
+        e1 = edges.CompoundEdge(self, "eF", (hy-t, h-hy+t))
+        e2 = edges.CompoundEdge(self, "Fe", (h-hy+t, hy-t))
+        e_back = ("F", e1, "F", e2)
+
+        p = self.edges["o"].settings.pin_height
+        e_inner_1 = edges.CompoundEdge(self, "fe", (y-p, p))
+        e_inner_2 = edges.CompoundEdge(self, "ef", (p, y-p))
+        e_inner_topbot = (e_inner_1, "f", e_inner_2, "f")
+
+        self.ctx.save()
+
```

Comment:
Mixing `y` and `x` wide parts in the first column is not ideal. This also breaks if y is smaller than x. Either rotate the parts or render them in the right `x` wide column.

Reply:
Good catch! The defaults create a square box, so I didn't notice it when testing. Fixed.